starter-code/src/clue.js

```diff
@@ -1,94 +1,176 @@
-description:  Billionare video game designer
-age:          22
-image:        https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg
-occupation:   Designer
+  // Characters Collection
+  var charactersArray = [mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard];
 
-missScarlet
-first_name:   Kasandra
-last_name:    Scarlet
-color:        red
-description:  She is an A-list movie star with a dark past
-age:          31
-image:        https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg
-occupation:   Actor
+  // Rooms' Collection
+  var roomsArray = [rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol];
 
-mrsPeacock
-first_name:   Eleanor
-last_name:    Peacock
-color:        blue
-description:  She is from a wealthy family and uses her status and money to earn popularity
-age:          36
-image:        https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg
-occupation:   Socialité
+  // Weapons Collection
+  var weaponsArray = [dinningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio];
 
-mrMustard
-first_name:   Jack
-last_name:    Mustard
-color:        yellow
-description:  He is a former football player who tries to get by on his former glory
-age:          62
-image:        https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg
-occupation:   Retired Football player
+  function randomSelector(arr) {
+  
+    var ranNumber = Math.floor(Math.random() * arr.length);
+    return arr[ranNumber];
 
-// Weapons
+  }
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+  var misteryEnvelope = []
 
-// Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+  function pickMistery () {
+    var cardCharacter = randomSelector(charactersArray);
+    var cardRooms = randomSelector(roomsArray);
+    var cardWeapons = randomSelector(weaponsArray);
+    var misteryEnvelope = [cardCharacter, cardWeapons, cardRooms];
+    return misteryEnvelope;
+  }
 
-// Characters Collection
-var charactersArray = [];
+  function revealMistery (misteryEnvelope) {
 
-// Rooms' Collection
-var roomsArray = [];
+    var firstName = misteryEnvelope[0].firstName;
+    var lastName = misteryEnvelope[0].lastName;
+    var weapon = misteryEnvelope[1].name;
+    var place = misteryEnvelope[2].name;
 
-// Weapons Collection
-var weaponsArray = [];
+   return firstname + " " + lastName + "killed Mr.Boddy using the " + weapon + " in the " + place + " !!!"
```

Comment:
should be firstName instead of firstname.
```
return "" + firstName + " " + lastName + " killed Mr.Boddy using the " + weapon + " in the " + place + "!!!"
```
Don't worry about the last error. It works in repl.it and the code is solid :)